recipes-core/install-on-emmc/files/install-on-emmc.sh

```diff
@@ -119,18 +119,23 @@ fi
 
 blink ${LED_GREEN} 0.25
 
+echo "Destroying the old GPT of eMMC ${EMMC_DEV}..."
+sgdisk -Z ${EMMC_DEV}
+
 # Calculate number of sectors to write:
 # Start of boot partition + sectors in that partition
 SECTORS="$(($(sfdisk -d ${BOOT_DEV} 2>/dev/null | tail -1 | sed 's/.*start=[[:space:]]*\([^,]*\), size=[[:space:]]*\([^,]*\).*/\1+\2/')))"
 
 echo "Writing ${SECTORS} sectors to eMMC ${EMMC_DEV}..."
 dd if=${BOOT_DEV} of=${EMMC_DEV} count=${SECTORS} conv=fsync
-partprobe ${EMMC_DEV}
+# fix the possible GPT header problem if BOOT_DEV is bigger than EMMC_DEV
+sgdisk -e ${EMMC_DEV}
 
 echo "Updating partition UUID of eMMC rootfs"
-partx -a ${EMMC_DEV}
-udevadm settle
 if ! test -b ${EMMC_DEV}p1; then
+	partprobe ${EMMC_DEV}
+	partx -a ${EMMC_DEV}
```

Comment:
what's the difference partprobe  and partx -a?
seems they are doing similar work

Reply:
Not sure, this is the old code, I just move the `partprobe ${EMMC_DEV}` from outside the `if` to inside.

And git blame shows it's you that added this code: [24d8479edb86da8e94dc17c315d3e85748cc04fd](https://github.com/siemens/meta-iot2050/commit/24d8479edb86da8e94dc17c315d3e85748cc04fd)

Reply:
OK, could remove one of this

Reply:
I am afraid these two were added to improve some stability, so better to not remove any of them. We can think better in the new solution of a flash sticker.